### PR TITLE
solidity: use modern cmake syntax

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/s/solidity.rb [hunk 1]
```diff
@@ -23,16 +23,16 @@ class Solidity < Formula
   depends_on "cmake" => :build
   depends_on xcode: ["11.0", :build]
   depends_on "boost"
+  depends_on "z3"
```
Comment (carlocab) [inline Formula/s/solidity.rb:26]: Can we use Homebrew `fmt` too?

**Review by carlocab — APPROVED** [review]

File: Formula/s/solidity.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/s/solidity.rb:26]: yeah, let me update the cmake to use the brewed fmt/nlohmann_json,range-v3

**Review by carlocab — APPROVED** [review]
Thanks! Surprised to not see linkage to `fmt`.

**Comment by chenrui333** [comment]
> Thanks! Surprised to not see linkage to `fmt`.

yeah, there is no linkage for it

**Pull request merged**